osquery/core/process.h

```diff
@@ -41,6 +41,9 @@ namespace osquery {
 /// Constant for an invalid process
 const auto kInvalidPid = (PlatformPidType)-1;
 
+/// The saved thread ID for shutdown to short-circuit raising a signal.
+static std::thread::id kMainThreadId;
```

Comment:
Ah. This needs to be `extern`, as we're intending to open a handle to it and wait on the thread to exit.